### PR TITLE
Fixes a bug in update when using mirrors

### PR DESCRIPTION
**Pull request by castulo**
There are a couple of problems when the user has a mirror set up,
and tries to update. The first one is that in the case where the
user is using the -p option and a mirror, the version of the mirror
was failing to be retrieved since the path_prefix was being added
to the path twice. The second problem was that when the mirror being
used by the user was stale, the mirror was being unset but some values
had already being cached, causing the update function to fail.

This PR fixes the issues and add a few tests to validate the correct behavior.

### CODE REVIEW

**Comment by castulo** [comment]
These bugs where found while trying to reproduce bug #669

File: test/functional/verify/verify-fix.bats [hunk 1]
```diff
@@ -143,16 +143,18 @@ test_setup() {
 		.deleted
 		--picky removing extra files under .*/target-dir/usr
 		REMOVING /usr/untracked_file3
-		Inspected 19 files
+		REMOVING /usr/share/defaults/swupd/versionurl
```
Comment (otaviobp) [inline test/functional/verify/verify-fix.bats]: All tests with changes on this commit should be in the 1st commit of this series, right?

Comment (castulo) [reply test/functional/verify/verify-fix.bats]: > All tests with changes on this commit should be in the 1st commit of this series, right?

ooh you are right, I'll move those to the first commit

**Comment by castulo** [comment]
yes @otaviobp we don't have tests for those flags, it'll be nice to add those, and I prefer to have explicit tests for them than "testing" them as a side effect of using them anyway

**Comment by castulo** [comment]
@otaviobp I moved the changes on the affected tests to the first commit as requested

**Pull request merged**